src/main/java/at/hannibal2/skyhanni/features/inventory/ReforgeHelper.kt

```diff
@@ -0,0 +1,417 @@
+            reforgeToSearch = null
+        }
+        itemToReforge = newItem
+        val newReforgeName = itemToReforge?.getReforgeName() ?: ""
+        if (newReforgeName == currentReforge?.lowercaseName) return
+        currentReforge = ReforgeAPI.reforgeList.firstOrNull { it.lowercaseName == newReforgeName }
+        updateDisplay()
+    }
+
+    @SubscribeEvent
+    fun onClick(event: GuiContainerEvent.SlotClickEvent) {
+        if (!isEnabled()) return
+        if (event.slot?.slotNumber == reforgeButton) {
+            if (event.slot.stack?.name == "§eReforge Item" || event.slot.stack?.name == "§cError!") return
+            if (currentReforge == reforgeToSearch) {
+                event.cancel()
+                waitForChat.set(false)
+                SoundUtils.playBeepSound()
+            } else if (waitForChat.get()) {
+                waitDelay = true
+                event.cancel()
+            } else {
+                if (event.clickedButton == 2) return
+                if (waitDelay) {
+                    waitDelay = false
+                } else {
+                    waitForChat.set(true)
+                }
+            }
+        }
+
+        DelayedRun.runNextTick {
+            itemUpdate()
+        }
+    }
+
+    @SubscribeEvent
+    fun onChat(event: LorenzChatEvent) {
+        if (!isEnabled()) return
+        when {
+            reforgeChatMessage.matches(event.message) -> {
+                DelayedRun.runDelayed(2.ticks) {
+                    itemUpdate()
+                    waitForChat.set(false)
+                }
+                if (config?.hideChat == true) {
+                    event.blockedReason = "reforge_hide"
+                }
+            }
+
+            reforgeChatFail.matches(event.message) -> {
+                DelayedRun.runDelayed(2.ticks) {
+                    waitForChat.set(false)
+                }
+                if (config?.hideChat == true) {
+                    event.blockedReason = "reforge_hide"
+                }
+            }
+        }
+    }
+
+    @SubscribeEvent
+    fun onOpen(event: InventoryFullyOpenedEvent) {
+        if (!LorenzUtils.inSkyBlock) return
+        when {
+            isHexReforgeMenu(event.inventoryName) -> {
+                isInHexReforgeMenu = true
+                DelayedRun.runDelayed(2.ticks) {
+                    itemUpdate() // update since an item must already be in place
+                }
+            }
+
+            isReforgeMenu(event.inventoryName) -> {
+                itemToReforge = null
+                currentReforge = null
+            }
+
+            else -> return
+        }
+        isInReforgeMenu = true
+        waitForChat.set(false)
+        DelayedRun.runNextTick {
+            inventoryContainer = Minecraft.getMinecraft().thePlayer.openContainer
+        }
+    }
+
+    @SubscribeEvent
+    fun onClose(event: InventoryCloseEvent) {
+        if (!isEnabled()) return
```

Comment:
when the inventory close happens while not in sb (for whatever reasons) then the values below will not get correctly reset. maybe move the is enabled check above updateDisplay and reset all values always regardless